www/src/components/shell/Shell.js

```diff
@@ -37,9 +38,11 @@ function Shell({ shell }) {
   const fitAddon = useMemo(() => new FitAddon(), [])
   const [terminalTheme] = useContext(TerminalThemeContext)
   const { fresh } = useOnboarded()
+  const [restart, setRestart] = useState(false)
 
   useEffect(() => {
     if (!xterm?.current?.terminal) return
+    if (restart) return
```

Comment:
does this actually restart the terminal?

Reply:
Actually, this line is so that this effect won't run twice in a row since `restart` is first set to true by this effect and then to false by another one a bit below. This way "restart" will actually happen only once.

Reply:
If you know a good way to trigger this only once, maybe using some other react feature, I'd love to do it differently. This is a bit of a workaround.